### ISSUE TITLE
PCC corruption: split_query_key_value_and_split_heads with sharded I/O and non-tile-aligned sequence dim

### ISSUE THREAD

**Issue by rpavlovicTT** [tenstorrent/tt-metal#41526]
## Summary

`ttnn.transformer.split_query_key_value_and_split_heads` produces garbage output (PCC ~0.08) when input is BLOCK_SHARDED and output is HEIGHT_SHARDED with a non-tile-aligned sequence dimension (197). The DRAM interleaved path (`nlp_create_qkv_heads`) produces correct results (PCC=1.0).

This is triggered by the tt-mlir greedy optimizer (opt_level=2) on ViT, which chooses sharded memory configs for this op. At opt_level=1 (DRAM interleaved), ViT PCC=0.998.

## Minimal Reproducer

```python
import torch
import ttnn

def pcc(a, b):
    a_flat = a.flatten().float()
    b_flat = b.flatten().float()
    return torch.corrcoef(torch.stack([a_flat, b_flat]))[0, 1].item()

device = ttnn.open_mesh_device(ttnn.MeshShape((1, 1)))
dram = ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM)

block_sharded_input = ttnn.MemoryConfig(
    ttnn.TensorMemoryLayout.BLOCK_SHARDED, ttnn.BufferType.L1,
    ttnn.ShardSpec(
        ttnn.CoreRangeSet([ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(5, 7))]),
        [224, 384], ttnn.ShardOrientation.ROW_MAJOR))

height_sharded_output = ttnn.MemoryConfig(
    ttnn.TensorMemoryLayout.HEIGHT_SHARDED, ttnn.BufferType.L1,
    ttnn.ShardSpec(
        ttnn.CoreRangeSet([ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 5))]),
        [448, 64], ttnn.ShardOrientation.ROW_MAJOR))

torch_input = torch.randn(8, 197, 2304, dtype=torch.bfloat16)
tt_input = ttnn.from_torch(torch_input, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                            device=device, memory_config=dram)

# Golden: DRAM interleaved → correct
q_gold, k_gold, v_gold = ttnn.transformer.split_query_key_value_and_split_heads(
    tt_input, None, num_heads=12, transpose_key=False, memory_config=dram)
q_gold_t = ttnn.to_torch(q_gold)

# Test: BLOCK_SHARDED input → HEIGHT_SHARDED output → corrupt
tt_sharded = ttnn.to_memory_config(tt_input, block_sharded_input)
q_test, k_test, v_test = ttnn.transformer.split_query_key_value_and_split_heads(
    tt_sharded, None, num_heads=12, transpose_key=False, memory_config=height_sharded_output)
q_test_t = ttnn.to_torch(ttnn.to_memory_config(q_test, dram))

print(f"Q PCC={pcc(q_gold_t, q_test_t):.6f}")  # Prints ~0.08 instead of ~1.0
ttnn.close_mesh_device(device)
```

## Results

| Path | Q PCC | K PCC | V PCC |
|------|-------|-------|-------|
| DRAM interleaved (golden) | 1.000 | 1.000 | 1.000 |
| BLOCK_SHARDED → HEIGHT_SHARDED | **0.082** | **-0.000** | **0.084** |

## Root Cause

When input is sharded, `split_query_key_value_and_split_heads` routes to `ttnn::experimental::create_qkv_heads()` (the sharded device kernel) instead of `nlp_create_qkv_heads()` (interleaved). The sharded kernel splits `[8, 197, 2304]` → `[8, 12, 197, 64]` Q/K/V, but the sequence dimension 197 is non-tile-aligned (197 % 32 = 5, padded to 224). The sharded kernel corrupts data when handling the padding across shard boundaries.

- Input: `[8, 197, 2304]` bf16 BLOCK_SHARDED on 6×8 grid, shard `[224, 384]`
- Output: `[8, 12, 197, 64]` bf16 HEIGHT_SHARDED on 8×6 grid (48 cores), shard `[448, 64]`
- Flattened output height: `8 × 12 × 224 = 21504`, `21504 / 48 = 448` per shard

## Impact

Blocks ViT model execution at opt_level=2 (greedy optimizer) in tt-xla. PCC drops from 0.998 (opt_level=1) to 0.057 (opt_level=2).

## Environment

- Device: wormhole_b0, n150
- tt-metal commit: `aa301f48dcd8e9f1b9057a21a79ffc981f551d9f`
- tt-mlir commit: `7157521b77ae11be1c13e6e9a879e73eb53c5547`
